### DISCUSSION TITLE
Instructions to upgrade with OLM

### DISCUSSION THREAD

**Discussion by laloyalo** [argoproj-labs/argocd-operator#284] [Q&A] [answered]
Hello everyone.

Thanks for the great work you've done with this operator. I see that finally the v0.15.0 version has been released with support for ArgoCD v2.0.0, which is amazing and thank you for that, I just installed it and I am testing it out.

I want to ask if there are any recommended steps to run an upgrade while using the OLM. Ever since I started using the operator through the OLM, it has been really difficult to update it, and I am not sure if I might be doing something wrong. I tried to upgrade to the v0.15.0 version, but I could only get it to install the version available in the registry hosted in [this quay repo](https://quay.io/repository/jmckind/argocd-operator-registry?tag=latest&tab=tags), the image has not been updated in the past 6 months.

On previous updates the operator has not been able to upgrade my ArgoCD deployment to a newer version, and I have only been able to do so if I replaced the registry image for a newer one.

I would like to know if there are any guides or documentation on how to properly configure the operator with the OLM to make full use of the automatic upgrades.

Also, to upgrade to the v0.15.0 version I had to uninstall the operator and install it manually with the provided instructions in the docs (great docs by the way!).

Thanks again for all the great work.

**Comment by wtam2018** [comment]
@laloyalo Thanks for your question.   
Here is the documentation on how to install the operator
https://argocd-operator.readthedocs.io/en/latest/install/olm/

Here is the new operator image repo.
https://quay.io/repository/argoprojlabs/argocd-operator?tab=tags

We are in the process of getting argocd operator v0.0.15 published to OperatorHub.

HTH
Thanks.

**Reply by dirien** [reply]
This is really good news. As we, have already an internal discussion with our Red Hat TAMS about this :) 

Perfect! looking forward!

**Reply by gonzolino** [reply]
That's really nice to hear :)

Do you have a timeline when we can expect v0.0.15 to be published on OperatorHub?

**Reply by Obirah** [reply]
Hi, this is also a pressing issue for us. Is something currently blocking the release of v0.0.15 to OperatorHub?

**Reply by BobClaerhout** [reply]
We also would like to update our argo-cd installation using the operator. Can you provide some insight on the timing which can be expected?

**Reply by iam-veeramalla** [reply]
@BobClaerhout currently operator installs argo cd v2.0.0. Which version of argo cd are you looking for ?

**Reply by Obirah** [reply]
@iam-veeramalla We're talking about the release of the operator v0.0.15 on OperatorHub: https://operatorhub.io/operator/argocd-operator - the version there is still v0.0.14

**Reply by dirien** [reply]
I am really interessted, what the possible reason is that we have such a delay in updateing the Operator in the OperatorHub.

We use Openshift 4, and there the [OperatorHub](https://docs.openshift.com/container-platform/4.6/operators/understanding/olm-understanding-operatorhub.html) function.

So it is curcial for us to get the newer version. Hope that helps to build up an understanding, why some people eagerly wait for this update 😄

**Reply by iam-veeramalla** [reply]
Hi @dirien @Obirah @BobClaerhout @gonzolino ,

 Currently @jomkz is running into some issue while publishing the operator into operator hub. @jomkz can you please share the issue and expected timeline for this :)

**Reply by Obirah** [reply]
Yeeaaaaaah, it's online now! Thank you!

**Reply by dirien** [reply]
Celebration! 💯 So cool!

**Reply by BobClaerhout** [reply]
Nice! Thanks!

**Comment by iam-veeramalla** [comment, marked as answer]
Closing the discussion as new version v0.1.0 is published.
